### PR TITLE
8341399: Add since checker tests to the langtools modules

### PR DESCRIPTION
**Pull request by nizarbenalla**
Can I get a review for this patch that adds `@since` checker tests to the following modules: java.compiler, jdk.compiler, jdk.javadoc and jdk.jdeps. The initial test for `java.base` has been integrated in [JDK-8331051](https://bugs.openjdk.org/browse/JDK-8331051). 

The jtreg comments are almost the same as the ones for the `java.base` test, only the Bug ID and the module name being passed to the test are different.

I've made a small change to `test/jdk/TEST.groups` to include the new tests.

Here are the emails  [[1](https://mail.openjdk.org/pipermail/jdk-dev/2024-June/009160.html)] [[2](https://mail.openjdk.org/pipermail/jdk-dev/2024-October/009474.html)] in `jdk-dev` describing how the tests work and how to run them.

TIA

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8341399](https://bugs.openjdk.org/browse/JDK-8341399): Add since checker tests to the langtools modules (**Sub-task** - P4)

### Reviewers
 * [Vicente Romero](https://openjdk.org/census#vromero) (@vicente-romero-oracle - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21542/head:pull/21542` \
`$ git checkout pull/21542`

Update a local copy of the PR: \
`$ git checkout pull/21542` \
`$ git pull https://git.openjdk.org/jdk.git pull/21542/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21542`

View PR using the GUI difftool: \
`$ git pr show -t 21542`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21542.diff">https://git.openjdk.org/jdk/pull/21542.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21542#issuecomment-2417338102)
</details>

### CODE REVIEW

File: test/jdk/tools/sincechecker/jdk_compiler/CheckSince_jdkCompiler.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8341399
+ * @summary Test for `@since` for java.base module
```
Comment (liach) [inline test/jdk/tools/sincechecker/jdk_compiler/CheckSince_jdkCompiler.java:27]: You forgot to change the summaries but remembered to change the `@run` directives. Curious!

Comment (nizarbenalla) [reply test/jdk/tools/sincechecker/jdk_compiler/CheckSince_jdkCompiler.java:27]: Nice catch! I'll update them

**Comment by liach** [comment]
Why are some test and the test for java.base in modules subdirectory, while some other jdk. module tests are not? Like `test/jdk/tools/sincechecker/modules/java_compiler/CheckSince_javaCompiler.java` vs `test/jdk/tools/sincechecker/jdk_compiler/CheckSince_jdkCompiler.java` (note `jdk_compiler` is not the a `modules` directory)

**Comment by nizarbenalla** [comment]
The tests are all meant to be under the `modules` directory, there was a problem with how subdirectories were shown in my IDE.

**Review by vicente-romero-oracle — APPROVED** [review]
lgtm

**Pull request closed without merging**